compile/ivy/src/main/scala/sbt/compiler/ComponentCompiler.scala

```diff
@@ -64,4 +94,123 @@ class ComponentCompiler(compiler: RawCompiler, manager: ComponentManager) {
       manager.define(binID, Seq(targetJar))
     }
   }
-}
\ No newline at end of file
+}
+
+private[compiler] class IvyComponentCompiler(compiler: RawCompiler, manager: ComponentManager, ivyConfiguration: IvyConfiguration, log: Logger) {
```

Comment:
Glad to see this has a name :).  Can you add a comment on what this class is meant to do at the top?